page.go

```diff
@@ -69,70 +72,40 @@ func (p page) Marshal() ([]byte, error) {
 
 	buffer := new(bytes.Buffer)
 
-	// write payloadSize and payload
-	err1 := binary.Write(buffer, binary.LittleEndian, uint64(len(p.payload)))
-	_, err2 := buffer.Write(p.payload)
+	// write checksum, pageStatus, transactionNumber and nextPage
+	_, err1 := buffer.Write(p.transactionChecksum[:])
+	err2 := binary.Write(buffer, binary.LittleEndian, p.pageStatus)
+	if p.pageStatus == pageStatusInvalid {
```

Comment:
Probably okay if this sanity check goes at the top of the function, so it doesn't clutter the actual marshalling code.